### ISSUE TITLE
Not fully rendering Map and markers

### ISSUE THREAD

**Issue by ximet** [KoRiGaN/Vue2Leaflet#157]
Have the problem with map: not fully rendering Map.

My application work using vuejs-templates and vue-loader. And I found some problem with webpack and Vue2Leaflet, after that I add to my main.js next lines:

```
import L from 'leaflet';
delete L.Icon.Default.prototype._getIconUrl;

L.Icon.Default.mergeOptions({
  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
  iconUrl: require('leaflet/dist/images/marker-icon.png'),
  shadowUrl: require('leaflet/dist/images/marker-shadow.png')
});
```

And after that, I have the same issue.

### My component:

```
<template>
  <div>
    <l-map style="height: 90%" :zoom="zoom" :center="center">
      <l-tile-layer :url="url" :attribution="attribution"></l-tile-layer>
      <l-marker :lat-lng="marker"></l-marker>
    </l-map>
  </div>
</template>

<script>
import { LMap, LTileLayer, LMarker } from 'vue2-leaflet';
export default {
  name: 'example',
  components: {
    LMap,
    LTileLayer,
    LMarker
  },
  data () {
    return {
      zoom:13,
      center: L.latLng(47.413220, -1.219482),
      url:'http://{s}.tile.osm.org/{z}/{x}/{y}.png',
      attribution:'&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
      marker: L.latLng(47.413220, -1.219482),
    }
  },
  methods: {
    mounted() {
        setTimeout(function() { window.dispatchEvent(new Event('resize')) }, 250);
    }
  }
}
</script>
```

### Result rendering map:

<img width="917" alt="screen shot 2018-04-15 at 3 01 01 pm" src="https://user-images.githubusercontent.com/3358977/38778209-6a7cd294-40be-11e8-92c0-acd28ad89257.png">

### Question:
 How it fix? @KoRiGaN do you have any ideas?

**Comment by KoRiGaN** [comment]
Hi @ximet,

You need to import the css from leaflet into your project.
```
import "leaflet/dist/leaflet.css"
```
Hope this helps,

Mickaël

**Comment by ximet** [comment]
@KoRiGaN I added this styles, and have the same issue

**Comment by nwpuhmz** [comment]
@ximet I added 
`import "leaflet/dist/leaflet.css";`
to my main.js ,then solved it.

**Comment by KoRiGaN** [comment]
Closing this issue as not comment for 3 weeks,
@ximet if you still have this issue feel free to reopen it.

Micka

**Comment by martinnaughton** [comment]
I have the exact same issue. I didnt have an issue until I did a big update since christmas. It might have been the new 1.0.1 version. Same issue in 1.0.1 and 1.0.1.

Some tiles show and are in different orders. Tried debugging it but does not show up any errors.

When I include the css file then now tiles show. There might be a conflict some where.
`import "leaflet/dist/leaflet.css";`

Vue2Leaflet: 1.0.1 and Leaflet.js: 1.3.1
https://github.com/martinnaughton/leafletbasic

I am I missing some in the setup?

**Comment by ambrt** [comment]
Same here as @martinnaughton above.

**Comment by ambrt** [comment]
I found the solution.

First add to main.js:
```
import L from 'leaflet'
import 'leaflet/dist/leaflet.css'

delete L.Icon.Default.prototype._getIconUrl

L.Icon.Default.mergeOptions({
  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
  iconUrl: require('leaflet/dist/images/marker-icon.png'),
  shadowUrl: require('leaflet/dist/images/marker-shadow.png')
})
```
then the thing is that `height` of map must be in pixels,em or vh.

```
 <l-map ref="map"
                      :zoom="zoom"
                      :center="center"
                      style="height:200px;">
                      <l-tile-layer :url="url"
                        :attribution="attribution"></l-tile-layer>
                      <l-marker :lat-lng="marker"></l-marker>
</l-map>
```

**Comment by martinnaughton** [comment]
I can confirm that fixed the issue when doing both changes. 

Not sure how the example works then with version 1.0.2?
https://github.com/KoRiGaN/Vue2Leaflet/blob/master/examples/src/components/Simple.vue

**Comment by sp00x** [comment]
Does anyone by chance know how to get the markers working when using codesandbox.io ? The 'require' statements to patch things up are resolving to the content of the full images..

`TypeError
Failed to execute 'fetch' on 'Window': Failed to parse URL from �PNG (png data here)`

**Comment by merigold** [comment]
In my case fix this:
`
@import 'leaflet/dist/leaflet.css
`
 and add this to component
```
mounted() {
        this.$refs.map.mapObject._onResize();
    }
```

**Comment by psr1981** [comment]
I made it work by doing these two changes -

1) added style="height:800px" in <l-map> 
2) added "import 'leaflet/dist/leaflet.css"  in component.

**Comment by RamizSami** [comment]
Hey, I am still getting this same issue even after importing CSS. I am trying the code from here: https://github.com/KoRiGaN/Vue2Leaflet/blob/master/examples/src/components/Simple.vue. How can I fix this?

![image](https://user-images.githubusercontent.com/15404125/46107988-5d865380-c1f6-11e8-8c58-27685c3ec684.png)

**Comment by lordfuoco** [comment]
This is usually the result of missing or not properly imported CSS. How are you importing the CSS? Can you provide us with code or even better a code pen?

**Comment by RamizSami** [comment]
Hey @lordfuoco this is my vue file:
https://dpaste.de/YNAd

**Comment by lordfuoco** [comment]
@RamizSami  Hi!
So `
<style scoped>`

Your style is scoped, your import is going to be scoped, and since leaflet is not part of your component the css is not going to reach it.

Solution: if is okay remove scoped from the style block.
If is not okay import the css in the js -> `import "leaflet/dist/leaflet.css";`

**Comment by lordfuoco** [comment]
Closing this since is solved

**Comment by ady642** [comment]
For it was a problem of height of the map. I put height:inherit and it works

**Comment by BillalOI** [comment]
leather inside vuetify component

<l-map ref="map" v-resize="onResize" :zoom="zoom" :center="center">
      <l-tile-layer :url="url"></l-tile-layer>
      <l-marker :lat-lng="center"></l-marker>
</l-map>

  methods: {
      onResize() {
          this.$refs.map[0].mapObject._onResize();
      },
}

problem solved for me

**Comment by BillalOI** [comment]
leaflet inside vuetify component

```
<l-map ref="map" v-resize="onResize" :zoom="zoom" :center="center">
   <l-tile-layer :url="url"></l-tile-layer>
   <l-marker :lat-lng="center"></l-marker>
</l-map>

methods: {
   onResize() {
   this.$refs.map[0].mapObject._onResize();
   },
}
```

problem solved for me

**Comment by mlstoppa** [comment]
Removing any height and width expressed in percentage worked for me too.

**Comment by uchedotphp** [comment]
i noticed the Vue2Leaflet doesnt render well with Tailwindcss in the project

**Comment by AlanGreyjoy** [comment]
Anyone else having this problem with vuetify?

**Comment by lordjack** [comment]
> 
> 
> I found that adding
> `@import 'http://cdn.leafletjs.com/leaflet-0.7.5/leaflet.css';`
> instead
> `@import 'leaflet/dist/leaflet.css`
> renders the map correctly.

This problem solved for me too

**Comment by ideenkreise** [comment]
I have tried all methos but, still having the same issue

**Comment by Laxisss** [comment]
i tried everything and cannot make this work.

i tried adding `import 'leaflet/dist/leaflet.css';` to component, to main.js...
i tried adding `@import 'http://cdn.leafletjs.com/leaflet-0.7.5/leaflet.css';` too App.vue style without scoped

**Comment by sajadabs** [comment]
> i tried everything and cannot make this work.
> 
> i tried adding `import 'leaflet/dist/leaflet.css';` to component, to main.js... i tried adding `@import 'http://cdn.leafletjs.com/leaflet-0.7.5/leaflet.css';` too App.vue style without scoped i tried using `style="height:800px;"` in l-map component i tried using a class in l-map component and using height property in css with px, vh ... I tried using onResize but got a `Cannot read properties of undefined (reading '_onResize')`
> 
> i don't know what to do... i'm not using vue 2, i'm using vue3 with this [package](https://github.com/vue-leaflet/vue-leaflet)
> 
> EDIT : i did it, honestly i don't know how... i cleaned every tests i made and imported the css as the very first import of the component. i don't know if that's it but it worked...

same issue

**Comment by vuhuynhsk8** [comment]
It was not fully loaded might be due to rendering problems. 
In my case, I tried to set timeout to load the l-map component after 2 seconds, then it could show the map as expected.

Map.vue 

```<template>
  <div class="main">
    <div
      class="map"
      v-if="isMapReady"
    >
      <l-map
        :zoom="zoom"
        :center="center"
      >
        <l-tile-layer
          :url="url"
          :attribution="attribution"
        ></l-tile-layer>
        <l-marker :lat-lng="markerLatLng"></l-marker>
      </l-map>
    </div>
  </div>
</template>

<script>
export default {
  components: {},
  data() {
    return {
      isMapReady: false,
      url: 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',
      attribution:
        '&copy; <a target="_blank" href="http://osm.org/copyright">OpenStreetMap</a> contributors',
      zoom: 13,
      center: [10.799547, 106.712026],
      markerLatLng: [51.504, -0.159],
    };
  },
  created() {
    this.initializeMap();
  },
  mounted() {},
  methods: {
    initializeMap() {
      setTimeout(() => {
        this.isMapReady = true;
      }, 2000);
    },
  },
};
</script>

<style lang="scss" scoped>
.main ::v-deep {
  .map {
    height: 66vh;
  }
}
</style>
```

**Comment by gjae** [comment]
> **TL;DR:** height of map must be in pixels,em or vh not in %
> 
> I found the solution.
> 
> First add to main.js:
> 
> ```
> import L from 'leaflet'
> import 'leaflet/dist/leaflet.css'
> 
> delete L.Icon.Default.prototype._getIconUrl
> 
> L.Icon.Default.mergeOptions({
>   iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
>   iconUrl: require('leaflet/dist/images/marker-icon.png'),
>   shadowUrl: require('leaflet/dist/images/marker-shadow.png')
> })
> ```
> 
> then the thing is that `height` of map must be in pixels,em or vh.
> 
> ```
>  <l-map ref="map"
>                       :zoom="zoom"
>                       :center="center"
>                       style="height:200px;">
>                       <l-tile-layer :url="url"
>                         :attribution="attribution"></l-tile-layer>
>                       <l-marker :lat-lng="marker"></l-marker>
> </l-map>
> ```

This work for me, thanks!
